### PR TITLE
Add validation rules to disable nullable elements in a list, when used inside `@node` or `@relationshipProperties`.

### PR DESCRIPTION
**Pull request by MacondoExpress**
As the title, this PR adds a validation rule to avoid a list of nullable elements present in the `@node` or `@relationshipProperties` types as they are not natively supported in Neo4j.

Test: 3905.test.ts is skipped as`@cypher` fields with non-`@node` target are currently not implemented in the 7.x branch

### CODE REVIEW

File: packages/graphql/src/schema/validation/custom-rules/valid-types/valid-list-in-node-type.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+/**
+ * Validates that list types used in type annotated with the node directive are supported by Neo4j
+ **/
+export function ValidListInNodeType(context: SDLValidationContext): ASTVisitor {
+    return {
+        ObjectTypeDefinition(objectTypeDefinitionNode: ObjectTypeDefinitionNode, _key, _parent) {
+            const { directives } = objectTypeDefinitionNode;
+
+            const nodeUsage = directives?.find((directive) => directive.name.value === nodeDirective.name);
```
Comment (angrykoala) [inline packages/graphql/src/schema/validation/custom-rules/valid-types/valid-list-in-node-type.ts:40]: `nodeUsage` is confusing. Maybe `isNode`?

File: packages/graphql/src/schema/validation/custom-rules/valid-types/valid-list-in-node-type.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            const relationshipPropertiesUsage = directives?.find(
```
Comment (angrykoala) [inline packages/graphql/src/schema/validation/custom-rules/valid-types/valid-list-in-node-type.ts:41]: Same here

File: packages/graphql/src/schema/validation/custom-rules/valid-types/valid-list-in-node-type.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+                (directive) => directive.name.value === relationshipPropertiesDirective.name
+            );
+            if (!directives || (!nodeUsage && !relationshipPropertiesUsage)) {
+                return; // Skip non-node types or relationshipProperties types
+            }
```
Comment (angrykoala) [inline packages/graphql/src/schema/validation/custom-rules/valid-types/valid-list-in-node-type.ts]: I think this is a bit more readable as 2 separate ifs. i.e. two cases we want to return

File: packages/graphql/src/schema/validation/custom-rules/valid-types/valid-list-in-node-type.ts [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+                            errorMsg,
+                        })
+                    );
+                }
+            });
+        },
+    };
+}
+
+type TypePath = (string | Kind.LIST_TYPE | Kind.NON_NULL_TYPE)[];
```
Comment (angrykoala) [inline packages/graphql/src/schema/validation/custom-rules/valid-types/valid-list-in-node-type.ts]: Nitpick (no need to change it) I find the array syntax less confusing when dealing with nested types:

```suggestion
type TypePath = Array<string | Kind.LIST_TYPE | Kind.NON_NULL_TYPE>;
```

File: packages/graphql/src/schema/validation/custom-rules/valid-types/valid-list-in-node-type.ts [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+function typeNodeToString(typeNode: TypeNode): string {
+    if (typeNode.kind === Kind.NON_NULL_TYPE) {
+        return `${typeNodeToString(typeNode.type)}!`;
+    } else if (typeNode.kind === Kind.LIST_TYPE) {
+        return `[${typeNodeToString(typeNode.type)}]`;
+    }
+    return typeNode.name.value;
+}
```
Comment (angrykoala) [inline packages/graphql/src/schema/validation/custom-rules/valid-types/valid-list-in-node-type.ts]: Doesn't this already exists somewhere? Could we reuse that utility or it is too much of a different context?

**Review by angrykoala — APPROVED** [review]

File: packages/graphql/src/schema/validation/custom-rules/valid-types/valid-list-in-node-type.ts (hunk 1 above)
Comment (MacondoExpress) [reply packages/graphql/src/schema/validation/custom-rules/valid-types/valid-list-in-node-type.ts:40]: The reason why I appended `Usage` is to hint that is not used as a boolean, as later it is used to collect the name of the directive.
I'll accept any suggestion here!

File: packages/graphql/src/schema/validation/custom-rules/valid-types/valid-list-in-node-type.ts (hunk 5 above)
Comment (MacondoExpress) [reply packages/graphql/src/schema/validation/custom-rules/valid-types/valid-list-in-node-type.ts]: You're right we have an utility for that!

**Pull request merged**